zomes/wiki/code/src/page.rs

```diff
@@ -95,75 +88,120 @@ pub fn page_def() -> ValidatingEntryType {
         })
         .filter_map(Result::ok)
         .collect();
-
-    hdk::api::update_entry(
-        Page::from(title.clone(), sections, Some(timestamp)).entry(),
-        &page_address,
+    let page_address = create_page_if_non_existent(title.clone(), timestamp.clone())?;
+    let new_page_entry = Page::from(title.clone(), sections_address, timestamp).entry();
+    let new_address = hdk::update_entry(new_page_entry, &page_address)?;
+    hdk::link_entries(
+        &holochain_anchors::anchor("wiki_pages".into(), title.clone())?,
+        &new_address,
+        "anchor->page",
+        "",
     )?;
-
     Ok(title)
 }
-pub fn update_page(sections: Vec<Address>, title: String, timestamp: u64) -> ZomeApiResult<String> {
-    let page_address = create_page_if_non_existent(title.clone())?;
-    hdk::api::update_entry(
-        Page::from(title.clone(), sections, Some(timestamp)).entry(),
+
+pub fn update_page(
+    sections: Vec<Address>,
+    title: String,
+    timestamp: String,
+) -> ZomeApiResult<String> {
+    let page_address = create_page_if_non_existent(title.clone(), timestamp.clone())?;
+    let new_address = hdk::update_entry(
+        Page::from(title.clone(), sections, timestamp).entry(),
         &page_address,
     )?;
+    hdk::link_entries(
+        &holochain_anchors::anchor("wiki_pages".into(), title.clone())?,
+        &new_address,
+        "anchor->page",
+        "",
+    )?;
     Ok(title)
 }
-pub fn get_page(title: String) -> ZomeApiResult<JsonString> {
-    utils::get_entry(Page::initial(title).entry().address())
-}
 
-pub fn get_titles() -> ZomeApiResult<Vec<String>> {
-    let anchor_address = utils::anchor("wiki_pages", "all_pages")?;
-    Ok(hdk::utils::get_links_and_load_type::<Page>(
-        &anchor_address,
-        LinkMatch::Exactly("anchor->Page".into()),
+pub fn get_page(title: String) -> ZomeApiResult<Page> {
+    let address = match hdk::get_links(
+        &holochain_anchors::anchor("wiki_pages".into(), title)?,
+        LinkMatch::Exactly("anchor->page"),
         LinkMatch::Any,
     )?
-    .into_iter()
-    .map(|page| page.title)
-    .collect())
+    .addresses()
+    .last()
+    {
+        Some(address) => Ok(address),
+        None => Err(ZomeApiError::Internal("This page no exist".to_string())),
+    }?
+    .clone();
+    hdk::utils::get_as_type(address)
+}
+
+pub fn get_titles() -> ZomeApiResult<Vec<String>> {
+    get_titles_filtered("".to_string())
 }
 
 pub fn get_titles_filtered(data: String) -> ZomeApiResult<Vec<String>> {
-    let anchor_address = utils::anchor("wiki_pages", "all_pages")?;
-    Ok(hdk::utils::get_links_and_load_type::<Page>(
+    let anchor_address = Entry::App(
+        holochain_anchors::ANCHOR_TYPE.into(),
+        holochain_anchors::Anchor {
+            anchor_type: "wiki_pages".to_string(),
+            anchor_text: None,
+        }
+        .into(),
+    )
+    .address();
+
+    let titles = hdk::utils::get_links_and_load_type::<holochain_anchors::Anchor>(
```

Comment:
We don't need a get_links_and_load_type, we have the titles of the pages in the tags if we do a get_links_with_options and filter the tags